content/languages/python/codewars-test/index.md

```diff
@@ -5,15 +5,59 @@ tags:
   - testing
 sidebar: "language:python"
 prev: /languages/python/
+next: /languages/python/authoring/
 ---
 
 # Python Codewars Test Framework
 
-Codewars currently uses a [custom test framework][test-framework-repo] to test Python.
+To run Python tests, Codewars currently uses a custom test framework, published and available in [this GitHub repository][test-framework-repo].
 
 ## Overview
 
-The test framework allows writing named groups of tests holding other named groups of tests, containing assertions.
+The test framework provides facilities to create named, hierarchical groups of tests, as well as individual test cases.
+
+Additionally, the framework provides a set of functions performing assertions on various conditions: equality, error handling, truthness, etc.
+
+
+## Module `codewars_test`
+
+All functions, decorators, and assertions provided by the framework are defined in `codewars_test` Python module.
+
+:::warning Deprecation
+Python runner for verions prior to Python 3.8 does not contain the `codewars_test` module. The testing framework is imported implicitly and aliased as `test` and `Test`. This behavior is deprecated and for the Python 3.8 kata the explicit import is required.
+:::
+
+## Organization of tests
+
+Tests in Python testing framework are composed from functions decorated with a set of Python decorators. All such functions are automatically discovered and run. The final result for every test block is determined by contained assertions, and reported along with its measured execution time.
+
+
+### Test groups: `@describe`
```

Comment:
Should be `@test.describe`/`@test.it`. Also below.

Reply:
Actually, it should be `@codewars_test.describe` and `@codewars_test.it`. `@test.describe` and `@test.it` would work only if `codewars_test` would be aliased as `test` with `import codewars_test as test`, but it's not required to be done this way.

Reply:
Well IMO it seems like it would be a pain doing `codewars_test.assert_equals` and etc. In all the _examples_ you always did like `@test.describe`/`@test.timeout`/etc, so you might as well be consistent.

Come to think of it, who knows who might do `from codewars_test import *` and eliminate all the `test.`s, but whatever...

Reply:
I prefixed all mentions of decorators as `@test.describe` and `@test.it`. I left mentions of `describe` or `it` sections/blocks/groups without prefixes though, because they are not actual Python names, but a logical constructs.

Is it better now?